tests/data_files/server1.req.sha1.ext

```diff
@@ -0,0 +1,17 @@
+-----BEGIN CERTIFICATE REQUEST-----
```

Comment:
These days we use the `Makefile` in the same directory as a form of executable documentation for how these files were generated, please add an entry there. (We still commit the files, as not all commands are deterministics, and this particular `Makefile` is _not_ part of our build system or test system, it's really just executable documentation.)

(But thanks for documenting the command used in the PR description, in the old days we were not doing that and it only caused pain later. So, now we have this Makefile which is easier that finding the commit the file was introduced, and going to the PR from there.)

Reply:
I did the change as requested but I have a question.
Basically I just followed the easiest path by adding a new line in the Makefile with the command line I used to generate the CSR. However I noticed that this is not very elegant since other CSRs are generated with an internal program (which is inside this repo).
Unfortunately this program does not support extensions flags so it must be modified/extended accordingly in order to do so. However, if this change is only required for a single test, I don't know if it's worth to add this change...

@mpg What do you suggest me to do?

Reply:
I think in this case we want the test file to be generated by another program that ours, as it gives stronger guarantees of correctness: if we generated the file using our own program then the test would only ensure that if our code is correct today, then it remains correct in the future, while if we generate it with openssl, the test says that if OpenSSL is correct, then so are we (today and in the future), which seems better to me. So I'd recommend you keep the line as it is but perhaps add a comment like "generate this with OpenSSL as a comparison point to test we're getting the same result".